### PR TITLE
Api changes: quota per cluster.

### LINKED ISSUE #1179: Adjust `neuro config show-quota` to show the quota for the active cluster
Once https://github.com/neuromation/platform-client-python/issues/1175 and https://github.com/neuromation/platform-api/issues/992 are ready, we will need to look up the cluster stats by `cluster_name` currently stored in `.nmrc`.

### PR DESCRIPTION
**Pull request by shagren**
closes #1179

### CODE REVIEW

File: neuromation/api/quota.py [hunk 1]
```diff
@@ -8,7 +8,6 @@
 
 @dataclass(frozen=True)
 class _QuotaInfo:
-    name: str
```
Comment (dalazx) [inline neuromation/api/quota.py]: we could still store a cluster name here.

File: neuromation/api/quota.py [hunk 2]
```diff
@@ -42,17 +41,23 @@ def __init__(self, core: _Core, config: Config) -> None:
...
-        cpu_time_spent=spent_cpu,
-        cpu_time_limit=limit_cpu,
-    )
+def _quota_info_from_api(payload: Dict[str, Any]) -> Dict[str, _QuotaInfo]:
+    clusters = payload["clusters"]
+    ret: Dict[str, _QuotaInfo] = {}
+    for cluster_payload in clusters:
+        cluster_name = cluster_payload["name"]
+        jobs_payload = cluster_payload["jobs"]
+        quota_payload = cluster_payload.get("quota", {})
```
Comment (dalazx) [inline neuromation/api/quota.py]: looks like `quota` is always present in the response: https://github.com/neuromation/platform-api/blob/115c3f07e02356aec14fb473f71b799095f934d7/platform_api/handlers/stats_handler.py#L96

Comment (dalazx) [reply neuromation/api/quota.py]: but might be `{}` an empty dict, yep.

File: neuromation/api/quota.py (hunk 1 above)
Comment (shagren) [reply neuromation/api/quota.py]: Why we need name here?
Dict key will be more useful for end-user

Comment (dalazx) [reply neuromation/api/quota.py]: `QuotaInfoFormatter` could unambiguously output the cluster name for example.

Comment (shagren) [reply neuromation/api/quota.py]: added

File: neuromation/api/quota.py (hunk 2 above)
Comment (shagren) [reply neuromation/api/quota.py]: There was test for missed quota before. Then I add this case. Can I safetly remove or better to leave for future?

Comment (dalazx) [reply neuromation/api/quota.py]: we can leave as is. up to you.

File: neuromation/cli/config.py [hunk 3]
```diff
@@ -54,11 +54,18 @@ def config() -> None:
...
     """
-    Print quota and remaining computation time.
+    Print quota and remaining computation time for active cluster.
     """
-    quota = await root.client._quota.get(user)
+    quotas = await root.client._quota.get(user)
+    cluster_name = root.client.config.cluster_name
+    if cluster_name not in quotas:
+        raise ValueError(
+            f"No quota information available for cluster {cluster_name}\n"
```
Comment (serhiy-storchaka) [inline neuromation/cli/config.py]:
```suggestion
            f"No quota information available for cluster {cluster_name}.\n"
```

File: neuromation/api/quota.py [hunk 4]
```diff
@@ -42,17 +42,24 @@ def __init__(self, core: _Core, config: Config) -> None:
...
+        ret[cluster_name] = _QuotaInfo(
+            cluster_name=cluster_name,
+            cpu_time_spent=float(
+                int(jobs_payload["total_non_gpu_run_time_minutes"]) * 60
```
Comment (serhiy-storchaka) [inline neuromation/api/quota.py]: This was in the current code, but why do we need to call `int()` and `float()`? Are not all these values already integers?

Comment (shagren) [reply neuromation/api/quota.py]: Good catch! Will change soon

Comment (shagren) [reply neuromation/api/quota.py]: @serhiy-storchaka , can you check latest implementation?

File: neuromation/api/quota.py [hunk 5]
```diff
@@ -42,17 +45,23 @@ def __init__(self, core: _Core, config: Config) -> None:
...
+        quota_payload = cluster_payload.get("quota", {})
+        cpu_time_spent = jobs_payload["total_non_gpu_run_time_minutes"]
+        gpu_time_spent = jobs_payload["total_gpu_run_time_minutes"]
+        cpu_time_limit = quota_payload.get("total_non_gpu_run_time_minutes", None)
+        gpu_time_limit = quota_payload.get("total_gpu_run_time_minutes", None)
+
+        ret[cluster_name] = _QuotaInfo(
+            cluster_name=cluster_name,
+            cpu_time_spent=cpu_time_spent * 60,
+            cpu_time_limit=cpu_time_limit * 60 if cpu_time_limit else None,
```
Comment (serhiy-storchaka) [inline neuromation/api/quota.py]: What if `cpu_time_limit` is 0?

File: neuromation/cli/formatters/config.py [hunk 6]
```diff
@@ -47,22 +47,22 @@ def __call__(self, quota: _QuotaInfo) -> str:
...
         else:
             quota_str += self.QUOTA_NOT_SET
         return f"{spent_str} ({quota_str})"
 
-    def _format_time(self, total_seconds: float) -> str:
+    def _format_time(self, total_seconds: int) -> str:
         # Since API for `GET /stats/users/{name}` returns time in minutes,
         #  we need to display it in minutes as well.
-        total_minutes = int(total_seconds // 60)
+        total_minutes = total_seconds // 60
```
Comment (serhiy-storchaka) [inline neuromation/cli/formatters/config.py]: It is not related to this PR, but I think the code below can be simplified.
```python
hours, minutes = divmod(total_minutes, 60)
return f"{hours:02d}h {minutes:02d}m"
```

**Comment by asvetlov** [comment]
What is the reason for float->int replacement?

**Comment by serhiy-storchaka** [comment]
The reason is that all values are integers and calling `float()` to convert from integers from the Web API is boring. But AFAIK `int` is considered a subtype of `float` in type annotations, so we can keep `float` in annotations and set fields to `int` value (or `float('inf'`)).

I have no preferences, both approaches (`float` and `Optional[int]`) work to me.

**Comment by asvetlov** [comment]
Python time has 2 representations usually: it is either one of `datetime` classes or `float` as returned by `time.time()` and others. `time.struct_time` has too cumbersome API, it's usage is limited from my experience.

That's why the previous version used `float`. I guess people are very familiar with datetime-as-float types.
Quota API is private now but I expect it will become public eventually after a stabilization period. This is the reason why I do care about used types.

File: neuromation/cli/formatters/config.py (hunk 6 above)
Comment (shagren) [reply neuromation/cli/formatters/config.py]: Sure, changed

**Comment by shagren** [comment]
@asvetlov, thnx for explanation. Reverted to float back.

**Comment by shagren** [comment]
@asvetlov can we merge red? Like as errors not related to this PR

**Pull request merged**